### PR TITLE
refactor(core): phone method refactor

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Replace phone signin method with sms

Per our offline discussion, align sms as the phone passwordless signInMethod naming. 

## Testing

UT case passed
test locally on demo app
@logto-io/eng

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: packages/ui/src/apis/register.ts [hunk 1]
```diff
@@ -15,23 +15,23 @@ export const register = async (username: string, password: string) => {
     .json<Response>();
 };
 
-export const sendPhonePasscode = async (phone: string) => {
+export const sendSMSPasscode = async (phone: string) => {
```
Comment (IceHe) [inline packages/ui/src/apis/register.ts:18]: camelcase `registerSendSmsPasscode`.

image

See Notion doc: [coding convention](https://www.notion.so/silverhand/Common-36449f7172514044967e5dbe537b87e9#3deb4db5eb7846b28ce2d61a376a0082)

Comment (IceHe) [inline packages/ui/src/apis/register.ts:18]: camelcase `registerSendSmsPasscode` and etc.

image

See Notion doc: [coding convention](https://www.notion.so/silverhand/Common-36449f7172514044967e5dbe537b87e9#3deb4db5eb7846b28ce2d61a376a0082)

**Review by IceHe — CHANGES_REQUESTED** [review]

File: packages/ui/src/apis/register.ts (hunk 1 above)
Comment (simeng-li) [reply packages/ui/src/apis/register.ts]: @IceHe updated PTAL

**Review by IceHe — APPROVED** [review]
lgtm

**Pull request merged**